apps/web/pages/api/logo.ts

```diff
@@ -67,13 +147,28 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
   if (!domains) throw new Error("No domains");
 
   const [subdomain] = domains;
-  const { appLogo, appIconLogo } = await getTeamLogos(subdomain);
+  const teamLogos = await getTeamLogos(subdomain);
 
-  const filteredLogo = parsedQuery?.icon ? appIconLogo : appLogo;
+  // Resolve all icon types to team logos, falling back to Cal.com defaults.
+  const type: LogoType = parsedQuery?.type && isValidLogoType(parsedQuery.type) ? parsedQuery.type : "logo";
+  const logoDefinition = logoDefinitions[type];
+  const filteredLogo = teamLogos[logoDefinition.source] ?? logoDefinition.fallback;
 
   const response = await fetch(filteredLogo);
   const arrayBuffer = await response.arrayBuffer();
-  const buffer = Buffer.from(arrayBuffer);
+  let buffer = Buffer.from(arrayBuffer);
+
+  // If we need to resize the team logos (via Next.js' built-in image processing)
+  if (teamLogos[logoDefinition.source] && logoDefinition.w) {
```

Comment:
Would this work to prevent importing even when it's not needed?

```suggestion
  if (teamLogos[logoDefinition.source] && logoDefinition.w) {
  const { detectContentType, optimizeImage } = await import("next/dist/server/image-optimizer");
```